package/MDAnalysis/lib/distances.py

```diff
@@ -1369,7 +1419,7 @@ def calc_bonds(coords1: Union[np.ndarray, 'AtomGroup'],
                coords2: Union[np.ndarray, 'AtomGroup'],
                box: Optional[np.ndarray] = None,
                result: Optional[np.ndarray] = None,
-               backend: str = "serial") -> np.ndarray:
+               backend: str = "distopia" if HAS_DISTOPIA else "serial" ) -> np.ndarray:
```

Comment:
I'm really not sure how I feel about this on a brand new addition, changing defaults based on an optional library can be quite an issue when dealing with reproducibility. For example you can write a script, give it to two folks and one of them doesn't install distopia, then you end up spending ages trying to debug why the two things don't match.

Reply:
For 2.x distopia should be opt-in, just to be on the safe side.

Reply:
Okie dokie makes sense. Thoughts on making it the default for 3.0 once it's a bit more battle tested ? I suppose we can decide later :)

Reply:
The over-cautious person in me agrees with @orbeckst but I also know that most of our methods don't offer optional backends which means you end up using the slow stuff by default. Hence being really unsure about what the best option is :/

I'll take a passive stance on this one and go with concensus. If we go for non-default then I'm also going to propose that we check all the analysis methods and check if we can add a keyword arguments to select backends?

Reply:
I'd be cautious with anything that can change results. I totally get (and want!) faster distance calculations for everyone. Being able to set the backend in analysis tools is a good idea. It will allow us to directly run analysis with either and compare. If we find that for typical cases the difference is small or machine-eps-like then we can consider changing the default earlier. Otherwise we have a safe and clean upgrade path.

Propagating the backend keyword would also be important for folks who run analysis on different archs — some that are supported and others that are not. They might want to ensure that exactly the same analysis backend is used everywhere.

Reply:
I raised #3946

Reply:
I reverted to default = `serial`. We can decide what we want to do for defaults with a bit more maturity in `distopia`.